### PR TITLE
DBACCESS-1261: Add metrics to rs consul calls

### PR DESCRIPTION
**Pull request by jlinrbx**
# What problem are we solving?
We currently don't have any visibility into how many requests we're making to consul

https://jira.rbx.com/browse/DBACCESS-1261

# How are we solving the problem?
Add metrics for requests made to consul

# Checks
Please check these off before promoting the pull request to non-draft status.
- [x] All CI checks are green.
- [x] I have reviewed the proposed changes myself.

### CODE REVIEW

**Comment by jlinrbx** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/lib.rs [hunk 1]
```diff
@@ -207,9 +216,11 @@ impl Consul {
     /// # Errors:
     /// [ConsulError](consul::ConsulError) describes all possible errors returned by this api.
     pub async fn read_key(&self, request: ReadKeyRequest<'_>) -> Result<Vec<ReadKeyResponse>> {
+        let method_name = "read_key";
```
[the remark this answers is not in the document]
Comment (jlinrbx) [reply src/lib.rs]: Updated

File: src/lib.rs [hunk 2]
```diff
@@ -101,6 +102,14 @@ quick_error! {
     }
 }
 
+lazy_static! {
+    static ref CONSUL_REQUESTS_TOTAL: prometheus::CounterVec = prometheus::register_counter_vec!(
```
[the remark this answers is not in the document]
Comment (jlinrbx) [reply src/lib.rs:106]: Synced up with Kush and added failure and duration metrics

File: src/lib.rs [hunk 3]
```diff
@@ -589,21 +672,38 @@ impl Consul {
...
+        record_request_metric_if_enabled(&method, request_name);
         let future = self.https_client.request(req);
 
+        let step_start_instant = Instant::now();
         let response = if let Some(dur) = duration {
             match timeout(dur, future).await {
-                Ok(resp) => resp.map_err(ConsulError::ResponseError)?,
-                Err(_) => return Err(ConsulError::TimeoutExceeded(dur)),
+                Ok(resp) => resp.map_err(ConsulError::ResponseError),
+                Err(_) => Err(ConsulError::TimeoutExceeded(dur)),
```
Comment (sfong-roblox) [inline src/lib.rs:684]:
```suggestion
                Err(_) => Err(ConsulError::TimeoutExceeded(dur)),
```
why no return now?

Comment (sfong-roblox) [reply src/lib.rs:684]: oh I guess metrics?

**Review by sfong-roblox — APPROVED** [review]

File: src/lib.rs (hunk 3 above)
Comment (jlinrbx) [reply src/lib.rs:684]: Yeah, returning here would prevent failure and duration metrics from being recorded if there's a TimeoutExceeded error

**Pull request merged**